src/util/array_element_from_pointer.h

```diff
@@ -0,0 +1,32 @@
+/*******************************************************************\
+
+Module: Util
+
+Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+#ifndef CPROVER_UTIL_ARRAY_ELEMENT_FROM_POINTER_H
+#define CPROVER_UTIL_ARRAY_ELEMENT_FROM_POINTER_H
+
+#include "std_expr.h"
+
+/// Generate statement using pointer arithmetic to access the element at the
+/// given index of a pointer array:
+/// `*(pointer + index)`
+/// Arrays are sometimes (always in JBMC) represented as a pointer to their
+/// first element. This is because we cannot know the type of an array of
```

Comment:
Suggest replace the second sentence with just `, especially when their length is uncertain`

Reply:
I replaced a part of the second sentence with that. :slightly_smiling_face: